02-set-up.Rmd

```diff
@@ -125,32 +109,43 @@ server management. Fortunately there are many tools designed to ease monitoring
 - On Linux, the shell command `top` displays key resource use figures for most
 distributions. `htop` and Gnome's **System Monitor** (`gnome-system-monitor`, see figure \@ref(fig:2-1)) are more refined alternatives which use command-line and graphical user interfaces respectively.
 A number of options such as `nethogs` monitor internet usage.
-- On Windows the **Task Manager** provides key information on RAM and CPU use by
-process. This can be started in modern Windows versions by typing `Ctrl-Alt-Del` or by clicking the task bar and 'Start Task Manager'.
 - On Mac the **Activity Monitor** provides similar functionality. This can be initiated
 from the Utilities folder in Launchpad.
+- On Windows the **Task Manager** provides key information on RAM and CPU use by
+process. This can be started in modern Windows versions by typing `Ctrl-Alt-Del` or by clicking the task bar and 'Start Task Manager'.
 
 #### Exercises {-}
 
 1. What is the exact version of your computer's operating system?
-2. Start an activity monitor then type and execute the following code. How do the results on your system compare to those presented in Figure \@ref(fig:2-1)? 
+2. Start an activity monitor then execute the following code chunk.
+In it `lapply()` (or its parallel version `mclapply()`) is used to *apply* a function, `median()`, over every column in the data frame object
+`X` (see Section \@ref(the-apply-family) for more on the 'apply family' of 
+functions).[The
+reason this works is that a data frame is really a collection of list objects.
+]
+How do the system output logs results on your system compare to those presented in Figure \@ref(fig:2-1)? 
 
     ```{r, eval=FALSE}
-    # Note: this will consume more than 2 GB RAM and may take several seconds to run depending on hardware
+    # Note: uses 2+ GB RAM and several seconds or more depending on hardware
     # 1: Create large dataset
     X = as.data.frame(matrix(rnorm(1e8), nrow = 1e7))
     # 2: Find the median of each column using a single core
     r1 = lapply(X, median)
     # 3: Find the median of each column using many cores
-    r2 = parallel::mclapply(X, median) # runs in serial on Windows
+    r2 = parallel::mclapply(X, median) 
     ```
 
+```{block type='rmdtip'}
+`mclapply` only works in parallel on Mac and Linux. In Chapter 7 you'll learn about an equivalent function `parLapply()` that works in parallel on Windows. 
+```
+
+
 3. What do you notice regarding CPU usage, RAM and system time, during and after each of the three operations?
 4. Bonus question: how would the results change depending on operating system?
 
 ```{r timings-tests, echo=FALSE, eval=FALSE}
 system.time(lapply(X, median))
-system.time(parallel::mclapply(X, median))
+system.time(parallel::parLapply(X, median))
```

Comment:
Already fixed here: https://github.com/csgillespie/efficientR/pull/167/commits/19a1f5ad46f03377bc6a362537edd71ff29288cf - note the new rmdtip (just changed to an rmdnote).